plugins/modules/hostgroup.py

```diff
@@ -182,9 +190,25 @@ def main():
                     module_params['organizations'] = [module_params['organization']]
         expected_puppetclasses = module_params.pop('puppetclasses', None)
         entity = module.run()
+
         if not module.desired_absent and 'environment_id' in entity:
             ensure_puppetclasses(module, 'hostgroup', entity, expected_puppetclasses)
 
+        ansible_roles = module_params.get('ansible_roles')
+        if not module.desired_absent and ansible_roles is not None:
+            desired_ansible_role_ids = [item['id'] for item in ansible_roles]
+            current_ansible_role_ids = [
+                item['id'] for item in module.resource_action(
+                    'hostgroups', 'ansible_roles', {'id': entity['id']},
+                    ignore_check_mode=True, record_change=False,
+                )
+            ] if old_entity else []
+            if set(current_ansible_role_ids) != set(desired_ansible_role_ids):
+                module.resource_action(
+                    'hostgroups', 'assign_ansible_roles',
```

Comment:
will that also unassign roles that are not on that list?

Reply:
There are only those two endpoints available (listing and setting).
But you are right (of course) and i should not just assume...